src/utils/resources/vmi/utils/cloud-init.ts

```diff
@@ -1,23 +1,38 @@
+import { load } from 'js-yaml';
+
 import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
 
+type CloudinitUserDataObject = {
+  user?: string;
+  password?: string;
+  passwd?: { users: { name: string }[] };
+};
+
 export const getCloudInitCredentials = (
   vmi: V1VirtualMachineInstance,
-): { user: string; password: string } => {
+): { user?: string; password?: string; users?: string[] } => {
   const cloudInitVolume = vmi?.spec?.volumes?.find(
     (volume) => volume?.cloudInitNoCloud || volume?.cloudInitConfigDrive,
   );
 
-  const cloudInit = cloudInitVolume?.cloudInitNoCloud || cloudInitVolume?.cloudInitConfigDrive;
+  // KubeVirt supports the cloud-init NoCloud and ConfigDrive data sources
+  const cloudInitDataSource =
+    cloudInitVolume?.cloudInitNoCloud || cloudInitVolume?.cloudInitConfigDrive;
 
-  const cloudInitValuesStrings = cloudInit?.userData
-    ?.split('\n')
-    ?.filter((row) => !row?.includes('#cloud-config'));
+  const userDataObject: CloudinitUserDataObject = load(cloudInitDataSource?.userData);
 
-  const password = cloudInitValuesStrings
-    ?.find((row) => row?.includes('password'))
-    ?.split(': ')?.[1];
+  if (userDataObject?.user || userDataObject?.password) {
+    return {
+      user: userDataObject?.user,
+      password: userDataObject?.password,
+    };
+  }
 
-  const user = cloudInitValuesStrings?.find((row) => row?.includes('user'))?.split(': ')?.[1];
+  if (userDataObject?.passwd?.users) {
+    return {
+      users: userDataObject?.passwd?.users?.map((userobject: { name: string }) => userobject?.name),
```

Comment:
{ name: string } - i think can be removed as the declaration should be in the function type

Reply:
👍🏽 Thanks I'll checked that. I think I just added { name: string } there because of some eslint warning I got or something like that.